src/torchmetrics/functional/classification/stat_scores.py

```diff
@@ -16,8 +16,453 @@
+    multidim_average: str = "global",
+    ignore_index: Optional[int] = None,
+) -> None:
+    """Validate tensor input."""
+    # Check that they have same shape
+    _check_same_shape(preds, target)
+
+    # Check that target only contains [0,1] values or value in ignore_index
+    unique_values = torch.unique(target)
+    if ignore_index is None:
+        check = torch.any((unique_values != 0) & (unique_values != 1))
+    else:
+        check = torch.any((unique_values != 0) & (unique_values != 1) & (unique_values != ignore_index))
+    if check:
+        raise RuntimeError(
+            "Detected the following values in `target`: {unique_values} but expected only"
+            " the following values {[0,1] + [] if ignore_index is None else [ignore_index]}."
+        )
+
+    # If preds is label tensor, also check that it only contains [0,1] values
+    if not preds.is_floating_point():
+        unique_values = torch.unique(preds)
+        if torch.any((unique_values != 0) & (unique_values != 1)):
+            raise RuntimeError(
+                "Detected the following values in `preds`: {unique_values} but expected only"
+                " the following values [0,1] since preds is a label tensor."
+            )
+
+    if multidim_average != "global" and preds.ndim < 2:
+        raise ValueError("Expected input to be atleast 2D when multidim_average is set to `samplewise`")
+
+
+def _binary_stat_scores_format(
+    preds: Tensor,
+    target: Tensor,
+    threshold: float = 0.5,
+    ignore_index: Optional[int] = None,
+) -> Tuple[Tensor, Tensor]:
+    """Convert all input to label format."""
+    if preds.is_floating_point():
+        if not torch.all((0 <= preds) * (preds <= 1)):
+            # preds is logits, convert with sigmoid
+            preds = preds.sigmoid()
+        preds = preds > threshold
+
+    preds = preds.reshape(preds.shape[0], -1)
+    target = target.reshape(target.shape[0], -1)
+
+    if ignore_index is not None:
+        idx = target == ignore_index
+        target = target.clone()
+        target[idx] = -1
+
+    return preds, target
+
+
+def _binary_stat_scores_update(
+    preds: Tensor,
+    target: Tensor,
+    multidim_average: str = "global",
+) -> Tensor:
+    """"""
+    sum_dim = [0, 1] if multidim_average == "global" else 1
+    tp = ((target == preds) & (target == 1)).sum(sum_dim).squeeze()
+    fn = ((target != preds) & (target == 1)).sum(sum_dim).squeeze()
+    fp = ((target != preds) & (target == 0)).sum(sum_dim).squeeze()
+    tn = ((target == preds) & (target == 0)).sum(sum_dim).squeeze()
+    return tp, fp, tn, fn
+
+
+def _binary_stat_scores_compute(
+    tp: Tensor, fp: Tensor, tn: Tensor, fn: Tensor, multidim_average: str = "global"
+) -> Tensor:
+    return torch.stack([tp, fp, tn, fn, tp + fn], dim=0 if multidim_average == "global" else 1).squeeze()
+
+
+def binary_stat_scores(
+    preds: Tensor,
+    target: Tensor,
+    threshold: float = 0.5,
+    multidim_average: str = "global",
```

Comment:
As above, I'd be rather verbose for a public API.

```suggestion
    multidim_average: Literal["global", "samplewise"] = "global",
```